### DISCUSSION TITLE
I am porting libgit2sharp to Qt / C++

### DISCUSSION THREAD

**Discussion by StevePunak** [libgit2/libgit2#6856] [Ideas]
Hey there. My name is Stephen Punak.

We had a need to perform a range of git activities in our CI/CD pipelines where our processes are mostly Qt. I started out by referencing the libgit2sharp native calls as a way of understanding the native libgit2 C language flow in an OO model. As it turned out, a direct port to C++ seemed to be the best option.
So, I am about half way through a complete port of libgit2sharp to Qt/C++. I'm wondering if there is interest by the maintainers of libgit2 in a Qt port in general, or even hosting that in the libgit2 repo?

Thanks,
Stephen Punak

**Comment by ethomson** [comment]
Hi @StevePunak - we wouldn't likely add C++ or Qt to the libgit2 repository _directly_. We do have a handful of language bindings (in their own repo) within the libgit2 organization (see, for example, https://github.com/libgit2/libgit2sharp), which I'm happy to consider for future work, but it feels like the wrong direction (IMO), as it suggests a level of "officialness".

(For example, one would need to resolve the differences between a new Qt/C++ binding and and https://invent.kde.org/libraries/libqgit2 - is one "more official" than the other, and why? I don't have answers to these questions yet.)

In any case, I'm excited to see you working on this. 🙏 At a minimum, we're happy to update our README to add a link to your project, and I'm happy to figure out how we can work together.

**Comment by StevePunak** [comment]
Edward:

Thanks for the response. I probably wasn't clear in my original post, for which I apologize.

My intent was to create this as a potential **peer** to libgit2sharp in the libgit2 GitHub, not to put it in the libgit2 repository directly (which I agree would be a very wrong direction ;). Much of the code is actually a straightforward port from libgit2sharp, so it seems somewhere in this group of repos would be a natural home once it is stable.

Hope this clears things up. If I'm posting in the wrong place, any direction to the correct maintainer to contact would be appreciated.

Regards,
SP

**Comment by StevePunak** [comment]
Hey there Edward.

I've published the first cut of my C++/Qt binding library to my personal GitHub at https://github.com/StevePunak/libgit2qt.
I'd be very interested in getting it moved to the main set of libgit2 repositories as a peer to libgit2sharp if you guys are interested.

Regards,
Stephen Punak
